sql/core/src/test/scala/org/apache/spark/sql/DataFrameTimeWindowingSuite.scala

```diff
@@ -27,177 +29,234 @@ class DataFrameTimeWindowingSuite extends QueryTest with SharedSparkSession {
   import testImplicits._
 
   test("simple tumbling window with record at window start") {
-    val df = Seq(
-      ("2016-03-27 19:39:30", 1, "a")).toDF("time", "value", "id")
-
-    checkAnswer(
-      df.groupBy(window($"time", "10 seconds"))
-        .agg(count("*").as("counts"))
-        .orderBy($"window.start".asc)
-        .select($"window.start".cast("string"), $"window.end".cast("string"), $"counts"),
-      Seq(
-        Row("2016-03-27 19:39:30", "2016-03-27 19:39:40", 1)
+    def doTest(df: DataFrame): Unit = {
```

Comment:
Nit: Shall we use a loop instead of defining functions..? e.g. `Seq(df1, df2).foreach...`

Reply:
OK